cmd/ui.go

```diff
@@ -78,6 +78,24 @@ func (ui *UI) getFeatureLeadTime(prMergedAtString string, commits Commits) strin
 	return formatDuration(ui.subtractTime(prMergedAt, prFirstCommittedAt))
 }
 
+// getFirstReviewToLastReview returns the first review to last review time, in
+// hours and minutes, for a given PR.
+//
+//   firstReviewToLastReview = lastReviewedAt - firstReviewedAt
+//
+func (ui *UI) getFirstReviewToLastReview(login string, reviews Reviews) string {
+	for _, review := range reviews.Nodes {
+		if review.Author.Login != login {
+			firstReviewedAt, _ := time.Parse(time.RFC3339, review.CreatedAt)
+			lastReviewedAt, _ := time.Parse(time.RFC3339, reviews.Nodes[len(reviews.Nodes)-1].CreatedAt)
```

Comment:
I think that as-is, this could pick up an author's review as the last review (I don't think that that's your intention). Also, the last review might be of a type other than `APPROVED`. So, I think we have to ensure that:

- `lastReviewedAt` isn't authored by the pull request author
- `lastReviewedAt` is based off a review with the `APPROVED` state

This test case should help validate that (asserting `1h0m` because we want the difference between Joker's `CHANGES_REQUESTED` and `APPROVED` reviews):

```go
func Test_getFirstReviewToLastReview_AuthorReviewLast(t *testing.T) {
	var reviews = Reviews{
		Nodes: ReviewNodes{
			{
				Author: Author{
					Login: "Joker",
				},
				CreatedAt: "2022-04-06T16:11:09Z",
				State:     "CHANGES_REQUESTED",
			},
			{
				Author: Author{
					Login: "Joker",
				},
				CreatedAt: "2022-04-06T17:11:09Z",
				State:     "APPROVED",
			},
			{
				Author: Author{
					Login: "Batman",
				},
				CreatedAt: "2022-04-06T18:11:09Z",
				State:     "COMMENTED",
			},
		},
	}

	uiWithWeekends := &UI{
		Calendar: &cal.BusinessCalendar{
			WorkdayFunc:      WorkdayAllDays,
			WorkdayStartFunc: WorkdayStart,
			WorkdayEndFunc:   WorkdayEnd,
		},
	}

	st.Assert(t, uiWithWeekends.getFirstReviewToLastReview("Batman", reviews), "1h0m")
}
```

Reply:
Thanks for your review! You're right, this should only count if the last review is not the author... and I didn't consider the `COMMENT` review, great thought. Thank you for the test case. I added this test case and an additional one with the last review being a comment from a reviewer.